theseus/core/tests/test_cost_function.py

```diff
@@ -185,3 +185,64 @@ def error_fn(optim_vars, aux_vars):
         weighted_error = cost_function.weighted_error()
         direct_error_computation = cost_weight_value * torch.ones(batch_size, 1)
         assert torch.allclose(weighted_error, direct_error_computation)
+
+
+def test_autodiff_cost_function_error_and_jacobians_shape_on_SO3():
```

Comment:
Do we have to test the shape (taken care already by another test) or only that the numerical value of the jacobians are correct?

Reply:
Numerical values are tested in previous PR #98.

Reply:
If we already have other tests in place to check that sparse gradient computation is correct, here we should mostly test that the full combination of `AutodiffCostFunction` + `project` is put together properly and gives the expected result in a few test cases.

Reply:
Added value tests.